test/e2e/simple_test.go

```diff
@@ -120,3 +122,90 @@ func TestFailingBuild(t *testing.T) {
 		t.Fatalf("watchBuild did not return expected error: %v", err)
 	}
 }
+
+func TestBuildLowTimeout(t *testing.T) {
+	clients := setup(t)
+
+	buildName := "build-low-timeout"
+	buildTimeout := "50s"
```

Comment:
Optional: `buildTimeout = 50*time.Second` then `Timeout: buildTimeout.String()` means you don't have to parse and handle the error down on line 169. Up to you, it's also fine like this.